lib/flight_configuration/base_dsl.rb

```diff
@@ -177,6 +220,106 @@ def transform(key, value)
       else
         value.send(transform)
       end
+    rescue
+      # NOTE: Ideally the error would be logged, however this can't be done
+      #       without forming a recursive loop
+      if active_errors?
+        config.errors.add(key.to_sym, type: :transform, message: 'failed to coerce the data type')
+      else
+        raise Error, "Failed to coerce attribute: #{key}"
+      end
+    end
+
+    def validate!(config)
+      # Use active errors instead
+      if active_errors?
+        validate_active_errors!(config)
+
+      # Attempt to use validate! instead
+      elsif config.respond_to?(:validate!)
+        config.validate!
+
+      # Otherwise raise a generic error if invalid
+      elsif config.respond_to?(:valid?) && !valid?
+        raise Error, <<~ERROR
+          Failed to validate the application's configuration
+        ERROR
+      end
+    end
+
+    def validate_active_errors!(config)
```

Comment:
Can we remove the `!`.  The bang is used to indicate that the method is a dangerous, less safe, or unsurprising *alternative* to an equivalent non-bang method.  That is we should only have `validate_active_errors!` if there is already a `validate_active_errors`.

Reply:
0381df32a91097be7b0be25d332bbb221bd8d32f